pkg/distributor/distributor_test.go

```diff
@@ -3624,11 +3773,39 @@ func prepare(t testing.TB, cfg prepConfig) ([]*Distributor, []mockIngester, []*p
 		})
 	}
 
+	if len(cfg.ingesterDataPerZone) != 0 {
+		populateIngesterData(t, ingesters, cfg.ingesterDataPerZone, cfg.ingesterDataTenantID)
+	}
+
 	t.Cleanup(func() { stopAll(distributors, ingestersRing) })
 
 	return distributors, ingesters, registries
 }
 
+func populateIngesterData(t testing.TB, ingesters []mockIngester, dataPerZone map[string][]*mimirpb.WriteRequest, tenantID string) {
```

Comment:
[nit] I suggest adding an "s" to "ingesters" given it populate data for multiple ones.

```suggestion
func populateIngestersData(t testing.TB, ingesters []mockIngester, dataPerZone map[string][]*mimirpb.WriteRequest, tenantID string) {
```